SignalMessaging/categories/UIColor+OWS.h

```diff
@@ -59,19 +64,66 @@ NS_ASSUME_NONNULL_BEGIN
+@property (class, readonly, nonatomic) UIColor *ows_wintergreenColor;
+@property (class, readonly, nonatomic) UIColor *ows_tealColor;
+@property (class, readonly, nonatomic) UIColor *ows_blueColor;
+@property (class, readonly, nonatomic) UIColor *ows_indigoColor;
+@property (class, readonly, nonatomic) UIColor *ows_violetColor;
+@property (class, readonly, nonatomic) UIColor *ows_plumColor;
+@property (class, readonly, nonatomic) UIColor *ows_taupeColor;
+@property (class, readonly, nonatomic) UIColor *ows_steelColor;
+
+#pragma mark - Conversation Colors (Tint)
+
+@property (class, readonly, nonatomic) UIColor *ows_crimsonTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_vermilionTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_burlapTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_forestTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_wintergreenTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_tealTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_blueTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_indigoTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_violetTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_plumTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_taupeTintColor;
+@property (class, readonly, nonatomic) UIColor *ows_steelTintColor;
+
+#pragma mark - Conversation Colors (Shade)
+
+@property (class, readonly, nonatomic) UIColor *ows_crimsonShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_vermilionShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_burlapShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_forestShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_wintergreenShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_tealShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_blueShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_indigoShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_violetShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_plumShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_taupeShadeColor;
+@property (class, readonly, nonatomic) UIColor *ows_steelShadeColor;
+
+#pragma mark - Conversation Colors
+
++ (nullable OWSConversationColors *)ows_conversationColorsForColorName:(NSString *)colorName
+    NS_SWIFT_NAME(ows_conversationColors(colorName:));
+
+// If the conversation color name is valid, return its colors.
+// Otherwise return the "default" conversation colors.
++ (OWSConversationColors *)ows_conversationColorsOrDefaultForColorName:(NSString *)conversationColorName
+    NS_SWIFT_NAME(ows_conversationColorsOrDefault(colorName:));
+
+@property (class, readonly, nonatomic) NSArray<NSString *> *ows_conversationColorNames;
 
 + (NSString *)ows_defaultConversationColorName;
++ (OWSConversationColors *)ows_defaultConversationColors;
```

Comment:
The "default" conversation colors should only be relevant if a conversation has an invalid (obsolete?) conversation color.

Reply:
> obsolete?

It was my understanding that we'd be migrating everyone to a modern color, so this really is just a fail-safe, right?

Reply:
Right.